exec/java-exec/src/test/java/org/apache/drill/exec/fn/impl/TestDateFunctions.java

```diff
@@ -43,6 +46,11 @@
 public class TestDateFunctions extends PopUnitTestBase {
     static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(TestDateFunctions.class);
 
+    @BeforeClass
+    public static void setupLocal() {
+        Locale.setDefault(new Locale("en", "US"));
```

Comment:
This change also affects other unit tests which will be executed after the tests in this class. 
So to avoid this, for each unit test which depends on the locale, we should:
1. preserve current locale
2. change locale to "en"
3. execute test (in the try block)
4. restore locale (in the finally block).

As the example, you may use test testConstantFolding_allTypes() from the class TestConstantFolding below.